test/e2e/kubevirt.go

```diff
@@ -540,7 +526,104 @@ passwd:
 			}
 
 		}
-		fcosVM = func(idx int, labels map[string]string, butane string) (*kubevirtv1.VirtualMachine, error) {
+
+		addressesFromStatus = func(vmi *kubevirtv1.VirtualMachineInstance) func() ([]string, error) {
```

Comment:
why does this return a function ? Can't you just return the addresses and error, and wrap it in a func call whenever required ? (e.g. eventually / consistently blocks )

IMO it just makes the code slightly more complex.

Reply:
It allow you to use Eventually without a closure 
```golang
Eventually(addressesFromStatus(vmi)).                               
                WithPolling(time.Second).                                       
                WithTimeout(10*time.Second).                                    
                Should(HaveLen(expectedNumberOfAddresses), step)       
```

Reply:
it's probably just me, but I prefer to have that shoved in my face since it enforces a good practice of knowing I'm passing it a function. 

This works, but I need to keep on my toes and double check every call out there. If I see the closure, I know it's correct. This way, I don't. 

... but this is opinionated.